### PR TITLE
Overhaul of project

### PR DESCRIPTION
**Pull request by nistath**
Everything works well and the docs have been rewritten with clarity. All that remains is adding the 3D printing design files.
We could add resin-wifi-connect to this, but running node on a Pi Zero does not seem like a nice idea.

### CODE REVIEW

**Comment by nistath** [comment]
I am currently waiting for resin-wifi-connect Rust rewrite to be released before integrating it into the project since the Node version will soon be deprecated.

File: Dockerfile.template [hunk 1]
```diff
@@ -1,77 +1,40 @@
...
+    vlc \
+  && apt-get clean && rm -rf /var/lib/apt/lists/*
 
 # Move to app dir
 WORKDIR /usr/src/app
 
-# Move package.json to filesystem
-COPY "$SOURCEFOLDER/app/package.json" ./
+# Uncomment if you want systemd
+ENV CONF_DIR=/data/config/ \
```
Comment (jbaldwinroberts) [inline Dockerfile.template]: Can you comment this line please, what configs get stored here?

File: Dockerfile.template [hunk 2]
```diff
@@ -1,77 +1,40 @@
...
-# Disable uv4l service - we will manually start it later
+# Disable uv4l_raspicam service which we'll manually start later
 RUN systemctl disable uv4l_raspicam
 
-## uncomment if you want systemd
-ENV INITSYSTEM on
-
 # Start app
-CMD ["/bin/bash", "/usr/src/app/start.sh"]
+CMD /bin/bash /usr/src/app/start.sh
```
Comment (jbaldwinroberts) [inline Dockerfile.template]: I think this should be changed back, the docker docs state "CMD should almost always be used in the form of CMD [“executable”, “param1”, “param2”…]"

File: README.md [hunk 3]
```diff
@@ -1,34 +1,50 @@
...
 
 - Sign up on [resin.io](https://dashboard.resin.io/signup)
-- go throught the [getting started guide](http://docs.resin.io/raspberrypi/nodejs/getting-started/) and create a new application called `resincam`
-- clone this repository to your local workspace
-- set these variables in the `Fleet Configuration` application side tab
-
-  - `RESIN_HOST_CONFIG_gpu_mem` = `160`
-  - `RESIN_HOST_CONFIG_disable_camera_led` = `1`
+- Select your Raspberry Pi's version and create a new application
+- Follow the instructions on downloading and flashing [resinOS](https://resinos.io) to your Pi
```
Comment (jbaldwinroberts) [inline README.md]: resinOS is our unmanaged offering, is that what you meant to put?

File: README.md [hunk 4]
```diff
@@ -1,34 +1,50 @@
...
 ## Getting started
 
 - Sign up on [resin.io](https://dashboard.resin.io/signup)
-- go throught the [getting started guide](http://docs.resin.io/raspberrypi/nodejs/getting-started/) and create a new application called `resincam`
-- clone this repository to your local workspace
-- set these variables in the `Fleet Configuration` application side tab
-
-  - `RESIN_HOST_CONFIG_gpu_mem` = `160`
-  - `RESIN_HOST_CONFIG_disable_camera_led` = `1`
+- Select your Raspberry Pi's version and create a new application
```
Comment (jbaldwinroberts) [inline README.md]: I think we can just link to the getting started guide as it was before

File: README.md [hunk 5]
```diff
@@ -1,34 +1,50 @@
...
+- Follow the instructions on downloading and flashing [resinOS](https://resinos.io) to your Pi
+- Clone this repository to your local workspace
+- Set these variables in the `Fleet Configuration` application side tab
+  - `RESIN_HOST_CONFIG_gpu_mem` = `196`
```
Comment (jbaldwinroberts) [inline README.md]: perhaps a link to our docs explaining what these variables do

File: README.md [hunk 6]
```diff
@@ -1,34 +1,50 @@
...
+  - Select your device from the `Devices` side tab
+  - Click the `Actions` side tab
+  - Click the `ENABLE PUBLIC DEVICE URL` button
+  - Note down your device's public URL!
+- When your device finishes updating visit your device's IP or public URL to watch your stream!
+  - You can find your device's IP on the Resin dashboard.
+- Further explore [UV4L's configuration options](https://www.linux-projects.org/uv4l/tutorials/) and edit [the config file](config/uv4l.conf)
+- Make sure to update your devices after every change you make! You can either:
+  - Commit your changes and do another `git push resin master`
+  - Learn more about the powerful [**resin-cli**](https://github.com/resin-io/resin-cli) and how you can iterate and deploy to your devices rapidly!
```
Comment (jbaldwinroberts) [inline README.md]: perhaps you could directly reference `resin sync` here as a new user will have no idea which part of the CLI to look at or why `resin sync` is better

File: README.md (hunk 5 above)
Comment (jbaldwinroberts) [reply README.md]: and use the same style as you have for the environment variables

File: config/uv4l.conf [hunk 7]
```diff
@@ -96,7 +97,7 @@ preview = 240
 # awb = auto
 # imgfx = none
 # metering = average
-rotation = 180
+# rotation = 0
```
Comment (jbaldwinroberts) [inline config/uv4l.conf:100]: it would be cool if you could override these settings with environment variables

File: docs/BoM.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
+## Bill of Materials
```
Comment (jbaldwinroberts) [inline docs/BoM.md:1]: Quite a few dead links in this BOM

File: config/uv4l.conf (hunk 7 above)
Comment (nistath) [reply config/uv4l.conf:100]: You can. I describe how to do it in the README. I can add that one manually though.

File: docs/BoM.md (hunk 8 above)
Comment (nistath) [reply docs/BoM.md:1]: Yep, the Raspberry Pi foundation edited a lot of them after I added them.

File: Dockerfile.template (hunk 1 above)
Comment (nistath) [reply Dockerfile.template]: UV4L's config file as well as a text overlay config file.

File: README.md (hunk 5 above)
Comment (nistath) [reply README.md]: Same style meaning?

**Review by jbaldwinroberts — APPROVED** [review]
LGTM

**Pull request merged**